parsec/remote_dep_mpi.c

```diff
@@ -289,14 +283,11 @@ remote_dep_dequeue_init(parsec_context_t* context)
     }
 
     if( -1 == context->comm_ctx ) {
-        MPI_Comm comm;
-        MPI_Comm_dup(MPI_COMM_WORLD, &comm);
-        context->comm_ctx = (intptr_t)comm;
+        MPI_Comm_size( MPI_COMM_WORLD, (int*)&(context->nb_nodes));
+        MPI_Comm_rank( MPI_COMM_WORLD, (int*)&(context->my_rank));
+        context->comm_ctx = (intptr_t)MPI_COMM_WORLD;
```

Comment:
PaRSEc will continue to use a copy of the communicator. The problem with the current approach in PaRSEC is that we initialize the comm engine with this communicator before providing an opportunity to the user to change it. While, this is not a big issue at small scale, it totally destroyed our performance at scale. So, the current patch changes how the communication engine is initialized. It become lazy until we need it, and then it creates the dup of the communicator it is supposed to use (because the user can now change this communicator between parsec_init and parsec_start).